File: shared/attestation/implementation/src/commonMain/kotlin/com/appotato/shared/attestation/implementation/NoOpAttestationTokens.kt
package com.appotato.shared.attestation.implementation

import com.appotato.shared.attestation.api.AttestationTokens

/**
 * What a build with no Firebase configuration gets. Every call reports "no token", which is the
 * honest answer — and callers already have to handle it, because a real device can fail the
 * integrity check for reasons that have nothing to do with configuration.
 */
internal class NoOpAttestationTokens : AttestationTokens {
    override fun token(onResult: (String?) -> Unit) {
        onResult(null)
    }
}

File: iosApp/iosApp/NoOpAttestationTokens.swift
import ComposeApp
import Foundation

/// What a build with no Firebase configuration gets, mirroring `NoOpTelemetry` and
/// `NoOpRemoteConfig`. Reporting "no token" is the honest answer, and callers already handle it —
/// a real device can fail attestation for reasons that have nothing to do with configuration.
final class NoOpAttestationTokens: NSObject, AttestationTokens {
    func token(onResult: @escaping (String?) -> Void) {
        onResult(nil)
    }
}
